### ISSUE TITLE
Add support for ListViews (or ControlLists)

### ISSUE THREAD

**Issue by Jojatekok** [picoe/Eto#276]
In WPF, ListView is a control which can host and present a list of controls to the user, thus, allowing developers to implement their own fancy control as a list item.

**Comment by Jojatekok** [comment]
@cwensley I could also use TableLayouts in order to achieve the same, but I cannot add items to it while running the application (even after invalidating the control, it just stays the same).

**Comment by cwensley** [comment]
Yes this would be a great feature!  My initial thoughts were to extend the existing ListBox to specify a cell, which with #266 would allow you to have lists (and grids) with completely custom controls.

As for the TableLayout, I have plans to make it react at runtime to content changes.  I'm not sure when exactly I'll get to this, but I'm going to be looking at it as part of #248 (which will be soon).  Depending on how difficult, it might make it in for that. 

Thanks for the great feedback!

**Comment by claunia** [comment]
Apparently ListView is how WPF implements #565

**Comment by michaldobrodenka** [comment]
This would be great. I need something like this, but in the end I decided to embed WPF list controls. Will have to implement it natively also on Mac :( Good thing is, it was very fast to make it in WPF. (What troubles me, when WPF list is embedded in ETO, mouse wheel does not work)

I've tried to generate 1000 rows of controls into layout but it was unusable slow with Eto and WPF also. With more than 100 rows, special list controls are required. 

I've also considered for lists:

-embedding Xamarin.Forms (Mac/WPF or GTK# on both platforms) into ETO, but it is very unmature on desktop (January 2018)
- making lists in GTK# and embed them for ETO in Win and MAC - could not find many tutorials, I don't have experience with it
